R/zchunk_L2042.resbio_input_irr_mgmt.R

```diff
@@ -36,96 +54,294 @@ module_aglu_L2042.resbio_input_irr_mgmt_DISABLED <- function(command, ...) {
+    # level 2 names.
+    L103.ag_Prod_Mt_R_C_Y_GLU %>%
+      replace_GLU(map = basin_to_country_mapping) ->
+      L103.ag_Prod_Mt_R_C_Y_GLU
+
+    L123.For_Prod_bm3_R_Y_GLU %>%
+      replace_GLU(map = basin_to_country_mapping) ->
+      L123.For_Prod_bm3_R_Y_GLU
+
+    # add actual region names to table A_bio_frac_prod_R for joining later
+    A_bio_frac_prod_R %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      A_bio_frac_prod_R
+
+
+    # Forestry and Mill residue bio
+
+    # The function, add_bio_res_params_For_Mill, takes a data frame and adds user specified parameters and
+    # then repeats the resulting data frame for all MODEL_YEARS to form the  AgResBio_source
+    # output table
+    add_bio_res_params_For_Mill <- function(df, residueBiomassProduction, massConversion, harvestIndex, erosCtrl, massToEnergy, waterContent){
+      df %>%
+        mutate(residue.biomass.production = residueBiomassProduction,
+               mass.conversion = massConversion,
+               harvest.index = harvestIndex,
+               eros.ctrl = erosCtrl,
+               mass.to.energy = massToEnergy,
+               water.content = waterContent) %>%
+        repeat_add_columns(tibble::tibble(year = MODEL_YEARS))  ->
+        df1
+      df1
+    } # end add_bio_res_params_For_Mill
+
+
+    # 1. Form a table of Forest Residue Biomass Paramters by region-glu-year
+    L123.For_Prod_bm3_R_Y_GLU %>%
+      # Set up identifying information to fill in with parameters, incl 2.
+      select(GCAM_region_ID, GCAM_commodity, GLU) %>%
+      distinct %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      mutate(AgSupplySector = GCAM_commodity,
+             AgSupplySubsector = paste(GCAM_commodity, GLU, sep = aglu.CROP_GLU_DELIMITER),
+             AgProductionTechnology = AgSupplySubsector) %>%
+      add_bio_res_params_For_Mill("biomass", aglu.AVG_WOOD_DENSITY_KGM3, aglu.FOREST_HARVEST_INDEX,
+                                  aglu.FOREST_EROSION_CTRL_KGM2, aglu.WOOD_ENERGY_CONTENT_GJKG,
+                                  aglu.WOOD_WATER_CONTENT) %>%
+      select(-GCAM_region_ID, -GCAM_commodity, -GLU) ->
+      L204.AgResBio_For
+
+
+    # 2. Form a table of global Mill Residue Biomass Paramters by year
+    A_demand_technology %>%
+      filter(supplysector == "NonFoodDemand_Forest") %>%
+      select(supplysector, subsector, technology) %>%
+      rename(sector.name = supplysector,
+             subsector.name = subsector) %>%
+      add_bio_res_params_For_Mill("biomass", aglu.AVG_WOOD_DENSITY_KGM3, aglu.FOREST_HARVEST_INDEX,
+                                  aglu.MILL_EROSION_CTRL_KGM2, aglu.WOOD_ENERGY_CONTENT_GJKG,
+                                  aglu.WOOD_WATER_CONTENT) ->
+      L204.GlobalResBio_Mill
+
+
+    # 3. Forestry and Mill residue supply curves
+    # First, the Forest and Mill information bound into a single data frame for processing.
```

Comment:
It does, around line 157 (in the version I'm now working on). I'll try to clean up the comment.